build.sh

```diff
@@ -0,0 +1,33 @@
+set -e 
+name="oval-graph"
+module="oval_graph"
+
+if [ "$1" != "" ]; then
+    new_version=$1
+    if [[ $new_version =~ ^[0-9]{1,}.[0-9]{1,}.[0-9]{1,}$ ]]; then
+        old_version=$(python3 setup.py --version)
+        git checkout master
+        git pull upstream master
+        changes=$(git status --porcelain=v1 | grep -q '^\(.M\|M.\)')
+        if [$changes == ""];then
```

Comment:
This won't work - when grep is executed with `-q`, there is no output even if the repository is not clean. You need to check for the return code of `grep` directly, for example by putting as the condition statement to that `if`. Just keep in mind that if `grep` returns success, it means that there were changes.